Mathlib/CategoryTheory/Conj.lean

```diff
@@ -0,0 +1,197 @@
+#align category_theory.iso.hom_congr_refl CategoryTheory.Iso.homCongr_refl
+
+/- Porting note: removed `@[simp]`; simp can prove this -/
+theorem homCongr_trans {X₁ Y₁ X₂ Y₂ X₃ Y₃ : C} (α₁ : X₁ ≅ X₂) (β₁ : Y₁ ≅ Y₂) (α₂ : X₂ ≅ X₃)
+    (β₂ : Y₂ ≅ Y₃) (f : X₁ ⟶ Y₁) :
+    (α₁ ≪≫ α₂).homCongr (β₁ ≪≫ β₂) f = (α₁.homCongr β₁).trans (α₂.homCongr β₂) f := by simp
+#align category_theory.iso.hom_congr_trans CategoryTheory.Iso.homCongr_trans
+
+@[simp]
+theorem homCongr_symm {X₁ Y₁ X₂ Y₂ : C} (α : X₁ ≅ X₂) (β : Y₁ ≅ Y₂) :
+    (α.homCongr β).symm = α.symm.homCongr β.symm :=
+  rfl
+#align category_theory.iso.hom_congr_symm CategoryTheory.Iso.homCongr_symm
+
+variable {X Y : C} (α : X ≅ Y)
+
+/-- An isomorphism between two objects defines a monoid isomorphism between their
+monoid of endomorphisms. -/
+def conj : End X ≃* End Y :=
+  { homCongr α α with map_mul' := fun f g => homCongr_comp α α α g f }
+#align category_theory.iso.conj CategoryTheory.Iso.conj
+
+theorem conj_apply (f : End X) : α.conj f = α.inv ≫ f ≫ α.hom :=
+  rfl
+#align category_theory.iso.conj_apply CategoryTheory.Iso.conj_apply
+
+@[simp]
+theorem conj_comp (f g : End X) : α.conj (f ≫ g) = α.conj f ≫ α.conj g :=
+  α.conj.map_mul g f
+#align category_theory.iso.conj_comp CategoryTheory.Iso.conj_comp
+
+@[simp]
+theorem conj_id : α.conj (𝟙 X) = 𝟙 Y :=
+  α.conj.map_one
+#align category_theory.iso.conj_id CategoryTheory.Iso.conj_id
+
+@[simp]
+theorem refl_conj (f : End X) : (Iso.refl X).conj f = f := by
+  rw [conj_apply, Iso.refl_inv, Iso.refl_hom, Category.id_comp, Category.comp_id]
+#align category_theory.iso.refl_conj CategoryTheory.Iso.refl_conj
+
+@[simp]
+theorem trans_conj {Z : C} (β : Y ≅ Z) (f : End X) : (α ≪≫ β).conj f = β.conj (α.conj f) :=
+  homCongr_trans α α β β f
+#align category_theory.iso.trans_conj CategoryTheory.Iso.trans_conj
+
+@[simp]
+theorem symm_self_conj (f : End X) : α.symm.conj (α.conj f) = f := by
+  rw [← trans_conj, α.self_symm_id, refl_conj]
+#align category_theory.iso.symm_self_conj CategoryTheory.Iso.symm_self_conj
+
+@[simp]
+theorem self_symm_conj (f : End Y) : α.conj (α.symm.conj f) = f :=
+  α.symm.symm_self_conj f
+#align category_theory.iso.self_symm_conj CategoryTheory.Iso.self_symm_conj
+
+/- Porting note: removed `@[simp]`; simp can prove this -/
+theorem conj_pow (f : End X) (n : ℕ) : α.conj (f ^ n) = α.conj f ^ n :=
+  α.conj.toMonoidHom.map_pow f n
+#align category_theory.iso.conj_pow CategoryTheory.Iso.conj_pow
+
+/-- `conj` defines a group isomorphisms between groups of automorphisms -/
+def conjAut : Aut X ≃* Aut Y :=
+  (Aut.unitsEndEquivAut X).symm.trans <| (Units.mapEquiv α.conj).trans <| Aut.unitsEndEquivAut Y
+set_option linter.uppercaseLean3 false in
+#align category_theory.iso.conj_Aut CategoryTheory.Iso.conjAut
+
+theorem conjAut_apply (f : Aut X) : α.conjAut f = α.symm ≪≫ f ≪≫ α := by
+  cases f; cases α; apply CategoryTheory.Iso.ext 
+  simp only [conjAut, Aut.unitsEndEquivAut, conj]; rfl 
```

Comment:
Lean could not see through these definitions.

Reply:
great!

Reply:
yes, `aesop_cat` works!